### PR TITLE
OEL-3325: Responsive inpage navigation.

### PR DESCRIPTION
**Pull request by tibi2303**
Jira issue(s):
- [D8AGE-???](https://webgate.ec.europa.eu/CITnet/jira/browse/D8AGE-???)

### CODE REVIEW

File: includes/pattern.inc [hunk 1]
```diff
@@ -162,6 +162,7 @@ function oe_bootstrap_theme_preprocess_pattern_featured_media(&$variables) {
  * Implements hook_preprocess_HOOK() for inpage navigation pattern.
  */
 function oe_bootstrap_theme_preprocess_pattern_inpage_navigation(&$variables): void {
+  $variables['inpage_navigation_dropdown_id'] = $variables['attributes']['id'] . '-dropdown' ?? Html::getUniqueId('bcl-inpage-navigation') . '-dropdown';
```
Comment (brummbar) [inline includes/pattern.inc]: We always need to run everything through getUniqueId().
```suggestion
  $variables['inpage_navigation_dropdown_id'] = Html::getUniqueId(($variables['attributes']['id'] ?? 'bcl-inpage-navigation') . '-dropdown');
```

Comment (brummbar) [reply includes/pattern.inc]: I checked locally and indeed this pops up 
image

**Review by brummbar — CHANGES_REQUESTED** [review]
It looks good, but two remarks:
* miss tests for the ID at least, possibly icon, and the non-presence of the d-none class.
* the mockups have a bottom border for the nav on mobile viewport.
* let's add a css in this theme to account for when the administration bar is present;
* let's add css to account for when the globan is present.

File: includes/pattern.inc (hunk 1 above)
Comment (tibi2303) [reply includes/pattern.inc]: done, all the points.

**Comment by brummbar** [comment]
Turns out the globan is not sticky on top so there's no conflict, since we cannot position an inpage nav before the header.
The code for the toolbar doesn't work for me:
image

**Comment by tibi2303** [comment]
can be reviewed again

File: resources/sass/overrides/_toolbar.scss [hunk 2]
```diff
@@ -16,6 +24,13 @@
   --bs-offcanvas-zindex: 5070;
 }
 
+body:has(#globan) {
+  .bcl-inpage-navigation,
+  .bcl-sidebar {
+    top: 28px;
+  }
+}
+
```
Comment (drishu) [inline resources/sass/overrides/_toolbar.scss]: this is not needed in the end

File: tests/src/Kernel/fixtures/markup_rendering_patterns/inpage_navigation.yml [hunk 3]
```diff
@@ -73,3 +73,36 @@ inpage_navigation_with_no_links:
       class: Drupal\Tests\oe_bootstrap_theme\PatternAssertion\InPageNavigationAssert
       expected:
         title: 'Page content'
+inpage_navigation_with_dropdown_id_and_id:
```
Comment (drishu) [inline tests/src/Kernel/fixtures/markup_rendering_patterns/inpage_navigation.yml:78]: dont think this is actually needed, nice the things are not new pattern fields, there is no actual new scenario, just add check to the assertBaseElements and we're fine

File: tests/src/PatternAssertion/InPageNavigationAssert.php [hunk 4]
```diff
@@ -23,6 +23,16 @@ protected function getAssertions(string $variant): array {
...
+      'id' => [
+        [$this, 'assertElementAttribute'],
+        '.bcl-inpage-navigation',
+        'id',
+      ],
+      'dropdown_id' => [
+        [$this, 'assertElementAttribute'],
+        '.dropdown-toggle',
+        'id',
+      ],
```
Comment (drishu) [inline tests/src/PatternAssertion/InPageNavigationAssert.php:35]: you can just do this in assertBaseElements

File: tests/src/PatternAssertion/InPageNavigationAssert.php [hunk 5]
```diff
@@ -31,8 +41,15 @@ protected function getAssertions(string $variant): array {
...
     $crawler = new Crawler($html);
+
     $inpage_nav = $crawler->filter('ul.nav-pills');
     self::assertCount(1, $inpage_nav);
+
+    $svg_icon = $crawler->filter('h2.bcl-heading svg');
+    self::assertCount(1, $svg_icon);
+
+    $base_element = $crawler->filter('.bcl-inpage-navigation');
+    self::assertStringNotContainsString('d-none', $base_element->attr('class'));
```
Comment (drishu) [inline tests/src/PatternAssertion/InPageNavigationAssert.php:52]: think the negative asset is not really needed just because we removed this utlitity cass

**Review by drishu — DISMISSED** [review]

File: resources/sass/overrides/_toolbar.scss (hunk 2 above)
Comment (tibi2303) [reply resources/sass/overrides/_toolbar.scss]: i talked with fra, globan can be sticky and we should keep.

File: tests/src/Kernel/fixtures/markup_rendering_patterns/inpage_navigation.yml (hunk 3 above)
Comment (brummbar) [reply tests/src/Kernel/fixtures/markup_rendering_patterns/inpage_navigation.yml:78]: That's correct for the base elements, but I would keep the scenario, just use custom values for both of them to see that the value passed is respected.

File: tests/src/PatternAssertion/InPageNavigationAssert.php (hunk 5 above)
Comment (tibi2303) [reply tests/src/PatternAssertion/InPageNavigationAssert.php:52]: miss tests for the ID at least, possibly icon, and the non-presence of the d-none class.

Fra added this point in the first review..

File: tests/src/PatternAssertion/InPageNavigationAssert.php (hunk 4 above)
Comment (brummbar) [reply tests/src/PatternAssertion/InPageNavigationAssert.php:35]: We actually can't, as the value changes and that method cannot access any values passed.

**Review by brummbar — APPROVED** [review]

**Pull request merged**